internal/component/database_observability/mysql/component.go

```diff
@@ -274,52 +287,19 @@ func (c *Component) Update(args component.Arguments) error {
 
 	c.args = args.(Arguments)
 
-	dbConnection, err := sql.Open("mysql", formatDSN(string(c.args.DataSourceName), "parseTime=true"))
-	if err != nil {
-		return err
-	}
-
-	if dbConnection == nil {
-		return errors.New("nil DB connection")
-	}
-	if err = dbConnection.Ping(); err != nil {
-		return err
-	}
-	c.dbConnection = dbConnection
-
-	rs := c.dbConnection.QueryRowContext(context.Background(), selectServerInfo)
-	if err = rs.Err(); err != nil {
-		level.Error(c.opts.Logger).Log("msg", "failed to query engine version", "err", err)
-		return err
-	}
-
-	var serverUUID, engineVersion string
-	if err := rs.Scan(&serverUUID, &engineVersion); err != nil {
-		level.Error(c.opts.Logger).Log("msg", "failed to scan engine version", "err", err)
-		return err
-	}
-
-	c.args.Targets = append([]discovery.Target{c.baseTarget}, c.args.Targets...)
-	targets := make([]discovery.Target, 0, len(c.args.Targets)+1)
-	for _, t := range c.args.Targets {
-		builder := discovery.NewTargetBuilderFrom(t)
-		if relabel.ProcessBuilder(builder, database_observability.GetRelabelingRules(serverUUID)...) {
-			targets = append(targets, builder.Target())
-		}
-	}
-
-	c.opts.OnStateChange(Exports{
-		Targets: targets,
-	})
-
 	for _, collector := range c.collectors {
 		collector.Stop()
 	}
 	c.collectors = nil
 
-	if err := c.startCollectors(serverUUID, engineVersion); err != nil {
+	if err := c.startCollectors(); err != nil {
 		c.healthErr.Store(err.Error())
-		return err
+		// Export at least the base target so the component can expose metrics
+		// even if the database connection is currently unavailable.
+		c.opts.OnStateChange(Exports{
+			Targets: []discovery.Target{c.baseTarget},
```

Comment:
Thanks for moving stuff with the metric. I think we will still end up here with baseTarget not having the `server_id` metric that we're adding with relabels. 
This is a regression as it will mean we produce multiple AS for this DB (one with the server_id, one without).

Reply:
Unless while this function is running the targets are not available?